### PR TITLE
Uses varson parsedir

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: lib/confi.js [hunk 1]
```diff
@@ -47,18 +39,15 @@ var load = function(options) {
...
   }
-
   config = aug(true, defaultConf, envConf, userConf);
   config.env = opts.env;
   config.CWD = cwd;
-
-  config.ENV = process.env || {};
+  // config.ENV = process.env || {};
 
   Handlebars.registerHelper('default', function(value, defaultVal) {
```
Comment (jgallen23) [inline lib/confi.js]: we should remove all handlebars stuff

File: lib/confi.js [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 var aug = require('aug');
-var cjson = require('cjson');
-var yaml = require('js-yaml');
 var cwd = process.cwd();
 var glob = require('glob');
 var Handlebars = require('handlebars');
-
+var varson = require('varson');
+var parseDir = require('parse-dir');
+var _ = require('lodash')
```
Comment (jgallen23) [inline lib/confi.js]: I don't see this being used

**Comment by jgallen23** [comment]
still missing updated tests

**Comment by orthagonal** [comment]
This is a comment.

File: lib/confi.js [hunk 3]
```diff
@@ -47,45 +39,15 @@ var load = function(options) {
...
-  var checks = 0;
-  var configString = JSON.stringify(config);
-
-  while (checks < 10 && /({{\w+)/i.test(configString)) {
-    checks++;
-    configString = Handlebars.compile(configString)(config);
-    config = JSON.parse(configString);
-  }
-
+  var configString = JSON.stringify(varson(config));
```
Comment (jgallen23) [inline lib/confi.js]: what is the stringify and parse for?

File: test/test-override-settings.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+var chai = require('chai');
+var assert = chai.assert;
+var should = chai.should();
+var confi = require("../")
+
+
+
+describe('confi.load', function(){
```
Comment (jgallen23) [inline test/test-override-settings.js]: .load is deprecated.  We should update these tests

File: lib/confi.js [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
 var path = require('path');
 var aug = require('aug');
-var cjson = require('cjson');
-var yaml = require('js-yaml');
 var cwd = process.cwd();
 var glob = require('glob');
-var Handlebars = require('handlebars');
+var varson = require('varson');
+var parseDir = require('parse-dir');
+var _ = require('lodash');
```
Comment (jgallen23) [inline lib/confi.js]: doesn't look like we are using lodash anywhere.  Is this needed?

**Comment by jgallen23** [comment]
I think you still need to update package.json to include the two new libraries too

File: test/test-override-settings.js (hunk 4 above)
Comment (jgallen23) [inline test/test-override-settings.js]: lets just remove this test file

File: tests/confi.test.js [hunk 6]
```diff
@@ -1,5 +1,4 @@
 var confi = require('../');
-
 exports.testDefault = function(t) {
```
Comment (jgallen23) [inline tests/confi.test.js]: this is the old test file, right?  should we remove it?

File: test/conf/plugin-default.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (jgallen23) [inline test/conf/plugin-default.json]: this file should be default-plugin.json.  We need to make sure the tests are correct.

**Comment by jgallen23** [comment]
it looks like the tests we are doing is really testing varson and not confi.  We need to test pulling in multiple files, pulling in based on NODE_ENV, etc

File: .gitignore [hunk 8]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
 node_modules
+/tests
```
Comment (jgallen23) [inline .gitignore]: why are tests ignored?

File: package.json [hunk 9]
```diff
@@ -10,10 +10,10 @@
   },
   "dependencies": {
     "aug": "0.0.5",
-    "cjson": "0.2.1",
     "glob": "^5.0.3",
     "handlebars": "^3.0.0",
```
Comment (jgallen23) [inline package.json]: dont need cjson anymore

File: package.json [hunk 10]
```diff
@@ -10,10 +10,10 @@
...
-    "js-yaml": "~2.0.5"
+    "varson": "^0.0.1",
+    "parse-dir" : "^0.0.2"
   },
   "devDependencies": {
     "nodeunit": "*"
```
Comment (jgallen23) [inline package.json]: we need to update devDeps

File: test/test-override-settings.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  });
+  it('can open the production env', function(done){
+    var config = confi({ env: "production"});
+    assert.equal(config.analytics.enabled, true);
+    assert.equal(config.analytics.profile, 'ga-xxx');
+    assert.equal(config.host, 'prod');
+    assert.equal(config.env, 'production');
+    done();
+  });
+  it('can open a user-specific env', function(done){
```
Comment (jgallen23) [inline test/test-override-settings.js]: lets kill this test

File: test/test-override-settings.js [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    console.log('This test won\'t pass unless your username is jga :/');
+    var config = confi({
+      env: 'dev',
+      userConfig: true
+    });
+    assert.equal(config.analytics.enabled, true);
+    done();
+  });
+  it('can open a user-specific env with a path', function(done){
```
Comment (jgallen23) [inline test/test-override-settings.js]: and this one

File: test/test-override-settings.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
+var chai = require('chai');
+var assert = chai.assert;
+var should = chai.should();
+var confi = require("../")
+
+
+
+describe('confi', function(){
```
Comment (jgallen23) [inline test/test-override-settings.js]: can you add a test for the different files: `default-plugins.yaml, default-blah.yaml` ?

**Comment by orthagonal** [comment]
@jgallen23 I think this one is ready to merge

File: lib/confi.js [hunk 14]
```diff
@@ -1,18 +1,15 @@
 var fs = require('fs');
 var path = require('path');
 var aug = require('aug');
-var cjson = require('cjson');
-var yaml = require('js-yaml');
 var cwd = process.cwd();
 var glob = require('glob');
```
Comment (jgallen23) [inline lib/confi.js]: is glob still being used?

**Pull request merged**